src/Network/AWS/ARN/States.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Network.AWS.ARN.States
+  ( StateMachine (..),
+    parseStateMachine,
+    _StateMachine,
+  )
+where
+
+import Data.Hashable (Hashable)
+import Data.Text (Text)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Lens.Micro.Pro (Prism', prism')
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+-- >>> import Lens.Micro.Pro ((^?))
+
+-- | An AWS State Machine, made of its name.
+--
+-- >>> "stateMachine:orderProcessor" ^? _StateMachine
+-- Just (StateMachine {name = "orderProcessor"})
+--
+-- @since 0.3.3.0
+newtype StateMachine = StateMachine
+  { name :: Text
+  }
```

Comment:
Step Functions supports aliases and versions for state machines, and its manual does obliquely call them "qualifiers" like in Lambda. Should we include `qualifier :: Maybe Text`?

Reply:
Thanks. Updated.